### ISSUE TITLE
eks module tries to force cluster recreation if no cluster_security_group_id specified

### ISSUE THREAD

**Issue by davidaah** [terraform-aws-modules/terraform-aws-eks#1892]
## Description

There seems to be a regression of a very old issue https://github.com/terraform-aws-modules/terraform-aws-eks/pull/798 where if 
`cluster_security_group_id` is not specified and creation is disabled, the EKS module will try and replace the cluster every time.

Technically, the EKS module documentation does indicate `cluster_security_group_id` is a required parameter if `create_cluster_security_group` is set to `false`, however it is an optional parameter in the dependent `aws_eks_cluster` module and it should not be required since EKS creates a baseline cluster security group on cluster creation no matter what.

I think this should be as simple as wrapping this line in `compact()` -- 
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/main.tf#L16 to restore the previous behavior of removing any empty string values

## Versions
- Terraform:
   - Terraform v1.1.4 on darwin_amd64
 - Provider(s):
   + provider registry.terraform.io/hashicorp/aws v3.74.3
   + provider registry.terraform.io/hashicorp/cloudinit v2.2.0
   + provider registry.terraform.io/hashicorp/tls v3.1.0
 - Module: 
   - 18.2.7

## Reproduction

- Run terraform plan/terraform apply
- After first apply, re-run terraform plan

```
      ~ vpc_config {
          ~ cluster_security_group_id = "sg-XXXXXX" -> (known after apply)
          ~ security_group_ids        = [ # forces replacement
              + "",
            ]
          ~ vpc_id                    = "vpc-YYYYY" -> (known after apply)
            # (4 unchanged attributes hidden)
        }
```

### Code Snippet to Reproduce
```
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "18.7.2"

  cluster_name                    = "test-cluster"
...

  create_cluster_security_group = false
  create_node_security_group    = false
...
}
```

## Expected behavior

Modules should not detect any changes

## Actual behavior

Module thinks something has changed and tries to destroy

**Comment by drunkirishcoder** [comment]
can confirm, ran into this bug too, and `compact` fixes it.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.10.1 :tada:
